Directory.Build.props

```diff
@@ -1,8 +1,18 @@
 <Project>
+    <PropertyGroup>
+        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
+        <LangVersion>latest</LangVersion>
+    </PropertyGroup>
```

Comment:
Personally I think it makes sense to add this config only for CI, and allow developers to comment out a code block locally (for a quick sanity check) without removing using statements and fixing all warnings before being able to run tests.

Reply:
That diagnostic doesn't seem to be warning severity by default (atleast on my machine in VSCode it's only a suggestion), but I get what you mean. So in the pic below, the unused using is no problem, but a variable that isn't used get complaints

![image](https://github.com/Altinn/app-lib-dotnet/assets/5425986/2e4b5032-e799-49b3-b997-16c68555fd44)

Personally I'm fine either way

Reply:
Pro (only CI):
* more flexibility to run before all issues are fixed
* No need for ! or other warning supressions to test while working
Con:
* One more reason to be surprised when CI comes back with failiure, but we have this with sonar blocking us from adding new warnings now, and I don't think it is much of a problem